src/hypertable_restrict_info.c

```diff
@@ -105,14 +107,14 @@ dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber
 }
 
 static bool
-dimension_restrict_info_closed_add(DimensionRestrictInfoClosed *dri, StrategyNumber strategy, Const *c)
+dimension_restrict_info_closed_add(DimensionRestrictInfoClosed *dri, StrategyNumber strategy, Datum datum)
 {
-	int64		value = partitioning_func_apply(dri->base.dimension->partitioning, c->constvalue);
+	int32		value = partitioning_func_apply(dri->base.dimension->partitioning, datum);
 
 	switch (strategy)
 	{
 		case BTEqualStrategyNumber:
-			dri->value = value;
+			dri->values = list_append_unique_int(dri->values, value);
```

Comment:
I think the list actually needs to be passed down all the way to `dimension_restrict_info_closed_add` since if you have two lists then you want the intersection of the two. Consider `WHERE device_id IN ('dev_1', 'dev_2') and device_id IN ('dev_2', 'dev_3')`

Reply:
OR would not land either clause in the restrict info I believe. You should double check. I think restrict info are only clause that can be anded together at the top level

Reply:
Actually that is one of the things we might want to address. If you have some boolean expression like above we will not add any restriction info so we would scan all the chunks. Shouldn't we try to expand boolean expressions?